orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -0,0 +1,712 @@
+
+class SimpleModelWithCrossEntropyLoss(onnxblock.Model):
+    def __init__(self):
+        super(SimpleModelWithCrossEntropyLoss, self).__init__()
+        self.loss = onnxblock.loss.CrossEntropyLoss()
+
+    def build(self, output_name):
+        return self.loss(output_name)
+
+
+class SimpleTrainingModelWithMSELoss(onnxblock.TrainingModel):
+    def __init__(self):
+        super(SimpleTrainingModelWithMSELoss, self).__init__()
+        self.loss = onnxblock.loss.MSELoss()
+
+    def build(self, output_name):
+        return self.loss(output_name)
+
+
+class SimpleTrainingModelWithCrossEntropyLoss(onnxblock.TrainingModel):
+    def __init__(self):
+        super(SimpleTrainingModelWithCrossEntropyLoss, self).__init__()
+        self.loss = onnxblock.loss.CrossEntropyLoss()
+
+    def build(self, output_name):
+        return self.loss(output_name)
+
+
+class SimpleModelWithBCEWithLogitsLoss(onnxblock.Model):
+    def __init__(self):
+        super(SimpleModelWithBCEWithLogitsLoss, self).__init__()
+        self.loss = onnxblock.loss.BCEWithLogitsLoss()
+
+    def build(self, output_name):
+        return self.loss(output_name)
+
+
+class SimpleTrainingModelWithBCEWithLogitsLoss(onnxblock.TrainingModel):
+    def __init__(self):
+        super(SimpleTrainingModelWithBCEWithLogitsLoss, self).__init__()
+        self.loss = onnxblock.loss.BCEWithLogitsLoss()
+
+    def build(self, output_name):
+        return self.loss(output_name)
+
+
+# Test utility methods
+
+
+def _get_onnx_model(torch_model, model_inputs):
+    model_outputs = torch_model(*model_inputs)
+    if isinstance(model_outputs, torch.Tensor):
+        model_outputs = [model_outputs]
+    dynamic_axes = {}
+    input_names = []
+    output_names = []
+    for i, model_input in enumerate(model_inputs):
+        input_name = f"input-{i}"
+        input_names.append(input_name)
+        dynamic_axes[input_name] = {}
+        for dim_idx in range(len(model_input.shape)):
+            dynamic_axes[input_name].update({dim_idx: f"{input_name}_dim{dim_idx}"})
+
+    for i, model_output in enumerate(model_outputs):
+        output_name = f"output-{i}"
+        output_names.append(output_name)
+        dynamic_axes[output_name] = {}
+        for dim_idx in range(len(model_output.shape)):
+            dynamic_axes[output_name].update({dim_idx: f"{output_name}_dim{dim_idx}"})
+
+    f = io.BytesIO()
+    torch.onnx.export(
+        torch_model,
+        model_inputs,
+        f,
+        input_names=input_names,
+        output_names=output_names,
+        opset_version=14,
+        do_constant_folding=False,
+        training=torch.onnx.TrainingMode.TRAINING,
+        dynamic_axes=dynamic_axes,
+        export_params=True,
+        keep_initializers_as_inputs=False,
+    )
+    return onnx.load_model_from_string(f.getvalue())
+
+
+def _to_numpy(tensor):
+    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
+
+
+def _get_models(device, N, D_in, H, D_out):
```

Comment:
Readability: Are there names for D_in, H etc that would make them more clear?

Reply:
This example was picked directly from pytorch repo, and so the test inherited the variables. Will try to update the names.

Reply:
Updated the names.